web/src/stream_topic_history.ts

```diff
@@ -191,6 +191,7 @@ export class PerStreamHistory {
 
         if (existing.count <= num_messages) {
             this.topics.delete(topic_name);
+            fetched_stream_ids.delete(this.stream_id);
```

Comment:
Hmm, but after we call `update_topic_last_message_id`, won't we have complete history for the stream again?

It seems like maybe we should instead just be deleting the `is_complete_for_stream_id(this.stream_id)` check next?

Reply:
That makese more sense since the message fetch query shoud be cheaper than stream history fetch.